scanners/downloaders.py

```diff
@@ -82,16 +82,16 @@ def authenticated_download_with_rtoken(url, dest, auth, proxy=None):
     session = requests.Session()
 
     # get a token
+    token = oauth2_get_token_from_rtoken(auth, proxy, session)
```

Comment:
Inside the oauth2_get_token_from_rtoken() function there's still another:

> if proxy:
>         proxy = {
>             "https": f"http://{proxy['proxyHost']}:{proxy['proxyPort']}",
>             "http": f"http://{proxy['proxyHost']}:{proxy['proxyPort']}",
>         }
> 
Although this can fix the issue by limiting the proxy variable change in the function scope, how about just deleting the one inside the function, letting  oauth2_get_token_from_rtoken() called after the proxy variable update as the original code?

Reply:
the issue is that, iirc, `anonymous_download`  can be called from elsewhere (at least in `zap_none` after a quick grep), so we can't modify what `anonymous_download` expects as proxy format without doing more changes.

Alternatively, we might improve the situation with renaming the proxy function parameter and the proxy variable into something more meaningful: 
e.g.: 
`def anonymous_download(url, dest=None, rapidast_proxy=None)` (and similarly for the other 2 functions)
and:
```
if rapidast_proxy:
  request_proxy = {...}
```
This way we can always have a reminder of what we're dealing with and it will be easier to avoid confusion.

Reply:
Ok. Thanks. then let's find a chance to refactor it at a later time.